### PR TITLE
Add parameters {user,group}_override_hiera_merge

### PR DESCRIPTION
**Pull request by anders-larsson**
Spec tests will fail for Puppet 4.3 since #73  hasn't been merged yet.

### CODE REVIEW

**Comment by anders-larsson** [comment]
@Phil-Friderici Made it into a loop since it will make it easier to extend it to go through other hiera_merge parameters that haven't been added yet.

File: spec/fixtures/hiera/hiera.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+:backends:
+  - yaml
+:yaml:
+  :datadir: 'spec/fixtures/hiera'
+:hierarchy:
+  - fqdn/%{fqdn}
+  - spectest/%{spectest}
```
Comment (Phil-Friderici) [inline spec/fixtures/hiera/hiera.yaml]: Better use common as default hierachy here, than you don't need to set the spectest fact in the tests as well.
(Repeated for better readability)

**Comment by Phil-Friderici** [comment]
The fixture files in your spec tests are not that big. If you use something like:
```it { should contain_file("#{v[:mergetitle]}").with_content(/#{v:[expected_content]}/) }```
we wouldn't need these fixture files. Btw: no need to expect the header in that case ;)

**Comment by Phil-Friderici** [comment]
> I avoided to create common because it will break previously created spec tests for (user|group)_override_entores since it will include data from Hiera.

Good point ! Wasn't aware of that.

**Comment by anders-larsson** [comment]
I'll close and reopen this pull request against the development branch.

**Pull request closed without merging**